gammapy/cube/make.py

```diff
@@ -118,11 +69,97 @@ def run(self, obs_list):
         maps: dict of stacked counts, background and exposure maps.
         """
         for obs in ProgressBar(obs_list):
-            self.process_obs(obs)
+            # First make cutout of the global image
+            try:
+                exclusion, cutout_slices = self.maps['exclusion'].make_cutout(
+                    obs.pointing_radec, 2 * self.offset_max, mode=self.cutout_mode
+                )
+            except PartialOverlapError:
+                # TODO: can we silently do the right thing here? Discuss
+                log.warning("Observation {} not fully contained in target image. Skipping it.".format(obs.obs_id))
+                continue
+
+            log.info('Processing observation {}'.format(obs.obs_id))
+            map_maker_obs = MapMakerObs(
+                obs=obs,
+                geom=exclusion.geom,
+                offset_max=self.offset_max,
+                exclusion=exclusion,
+            )
+            maps = map_maker_obs.run()
+
+            # Stack maps from this observation into the total maps
+            self.maps['counts'].data[cutout_slices] += maps['counts'].data
+            self.maps['exposure'].data[cutout_slices] += maps['exposure'].quantity.to(self.maps['exposure'].unit).value
+
+            # TODO: decide how to handle the multiple background maps in the interface
+            self.maps['background'].data[cutout_slices] += maps['background_fov'].data
 
-        self.maps = {
-            'counts_map': self.counts_map,
-            'background_map': self.background_map,
-            'exposure_map': self.exposure_map
-        }
         return self.maps
+
+
+class MapMakerObs(object):
+    """Make maps for a single IACT observation.
+
+    Parameters
+    ----------
+    obs : `~gammapy.data.DataStoreObservation`
+        Observation
+    geom : `~gammapy.maps.WcsGeom`
+        Reference image geometry
+    offset_max : `~astropy.coordinates.Angle`
```

Comment:
I would expect that the foV mask is an argument one can pass to `MapMakerObs`. Then you don't have to change it once we have a different approach to it, no?
I would create the mask in `MapMaker.run()`, so that `MapMaker` or its future equivalent is the only object that has to know about the method to apply to build the FoV mask.